pkg/kubelet/kubelet.go

```diff
@@ -779,20 +779,29 @@ func NewMainKubelet(kubeCfg *kubeletconfiginternal.KubeletConfiguration,
 	klet.statusManager = status.NewManager(klet.kubeClient, klet.podManager, klet)
 
 	if utilfeature.DefaultFeatureGate.Enabled(features.RotateKubeletServerCertificate) && kubeDeps.TLSOptions != nil {
-		var ips []net.IP
-		cfgAddress := net.ParseIP(kubeCfg.Address)
-		if cfgAddress == nil || cfgAddress.IsUnspecified() {
+		var (
+			ips   []net.IP
+			names []string
+		)
+
+		// If we successfully discovered addresses from the cloudprovider, use
+		// those. Otherwise, use the configured addressed if provided. Otherwise,
+		// make a best guess.
+		if len(ips) != 0 || len(names) != 0 {
```

Comment:
How could these ever not have `len() == 0`? The `var` block above creates zero length arrays.

Reply:
Maybe

```
if len(cloudIPs) !=0 || len(cloudNames) != 0 {
```

Reply:
Ya that's right. I'll add a test :)

Reply:
Hmm, this function seems to be untested.